src/server/scripts/poller.js

```diff
@@ -0,0 +1,274 @@
+import moment from 'moment';
+import raven from 'raven';
+
+import logging from '../logging';
+import { conf } from '../helpers/config';
+import db from '../db';
+import {
+  getGitHubRepoUrl,
+  parseGitHubRepoUrl
+} from '../../common/helpers/github-url';
+import requestGitHub from '../helpers/github';
+import { internalGetSnapcraftYaml } from '../handlers/launchpad';
+
+
+const logger = logging.getLogger('poller');
+raven.config(conf.get('SENTRY_DSN')).install();
+
+
+// Process all Repository (DB) models synchronously. Check for changes using
+// `checkSnapRepository` and if changed request a LP snap build and mark
+// it as 'updated'.
+export const pollRepositories = (checker, builder) => {
+  logger.info('GitHub Repository Poller ...');
+  var raven_client = raven.Client(conf.get('SENTRY_DSN'));
+
+  checker = checker || checkSnapRepository;
+  builder = builder || buildSnapRepository;
+
+  // XXX: meh! no ES6 import support, great library :-/
+  let AsyncLock = require('async-lock');
+  let pollRepoLock = new AsyncLock();
+
+  let repoDB = db.model('Repository');
+  return repoDB.fetchAll().then(function (results) {
+    logger.info(`Iterating over ${results.length} repositories.`);
+    results.models.forEach((repo) => {
+      pollRepoLock.acquire('PROCESS-REPO-SYNC', async () => {
+        const owner = repo.get('owner');
+        const name = repo.get('name');
+
+        // XXX skip repos already updated recently, create a new datetime
+        // column exclusively for the poller.
+        const last_polled_at = repo.get('polled_at') || repo.get('updated_at');
+
+        if (!repo.get('snapcraft_name')) {
+          logger.info(`${owner}/${name}: NO SNAPCRAFT.YAML`);
+          return;
+        }
+
+        if (!repo.get('store_name')) {
+          logger.info(`${owner}/${name}: NOT REGISTERED`);
+          return;
+        }
+
```

Comment:
LP would fail to upload the snap, but I am not even sure the UI would let you do that. Better to be on the safe side and skip repos in that state.